### PR TITLE
ENH:  Added opacity setting for 2D slice intersections

### PR DESCRIPTION
**Pull request by sjh26**
See: http://na-mic.org/Mantis/view.php?id=4334

### CODE REVIEW

**Comment by sjh26** [comment]
@lassoan @agirault

File: Libs/MRML/Widgets/qMRMLDisplayNodeWidget.cxx [hunk 1]
```diff
@@ -62,6 +62,8 @@ void qMRMLDisplayNodeWidgetPrivate::init()
   Q_Q(qMRMLDisplayNodeWidget);
   this->setupUi(q);
 
+  this->SliceIntersectionOpacitySlider->setValue(1.0);
```
Comment (lassoan) [inline Libs/MRML/Widgets/qMRMLDisplayNodeWidget.cxx]: Remove this lines and set default value in .ui file instead

File: Libs/MRML/Widgets/qMRMLDisplayNodeWidget.cxx [hunk 2]
```diff
@@ -252,6 +256,31 @@ void qMRMLDisplayNodeWidget::setSliceIntersectionThicknessVisible(bool visible)
 }
 
 //------------------------------------------------------------------------------
+void qMRMLDisplayNodeWidget::setSliceIntersectionOpacity(double opacity)
+{
+  Q_D(qMRMLDisplayNodeWidget);
+  if (!d->MRMLDisplayNode.GetPointer())
+  {
```
Comment (lassoan) [inline Libs/MRML/Widgets/qMRMLDisplayNodeWidget.cxx]: indentation

File: Libs/MRML/Widgets/qMRMLDisplayNodeWidget.cxx [hunk 3]
```diff
@@ -252,6 +256,31 @@ void qMRMLDisplayNodeWidget::setSliceIntersectionThicknessVisible(bool visible)
...
+void qMRMLDisplayNodeWidget::setSliceIntersectionOpacityVisible(bool visible)
+{
+  Q_D(qMRMLDisplayNodeWidget);
+  d->SliceIntersectionOpacitySlider->setVisible(visible);
+}
+
+//------------------------------------------------------------------------------
 void qMRMLDisplayNodeWidget::updateWidgetFromMRML()
 {
   Q_D(qMRMLDisplayNodeWidget);
```
Comment (lassoan) [inline Libs/MRML/Widgets/qMRMLDisplayNodeWidget.cxx]: Need to implement update of slice intersection opacity widget from MRML in updateWidgetFromMRML()

**Comment by sjh26** [comment]
Requested changes made and pushed!

File: Libs/MRML/Core/vtkMRMLDisplayNode.cxx [hunk 4]
```diff
@@ -322,6 +324,12 @@ void vtkMRMLDisplayNode::ReadXMLAttributes(const char** atts)
       ss << attValue;
       ss >> Opacity;
       }
+    else if (!strcmp(attName, "sliceIntersectionOpacity"))
+    {
```
Comment (lassoan) [inline Libs/MRML/Core/vtkMRMLDisplayNode.cxx]: indentation

File: Libs/MRML/Widgets/qMRMLDisplayNodeWidget.cxx [hunk 5]
```diff
@@ -252,6 +254,31 @@ void qMRMLDisplayNodeWidget::setSliceIntersectionThicknessVisible(bool visible)
 }
 
 //------------------------------------------------------------------------------
+void qMRMLDisplayNodeWidget::setSliceIntersectionOpacity(double opacity)
+{
+  Q_D(qMRMLDisplayNodeWidget);
+  if (!d->MRMLDisplayNode.GetPointer())
+    {
+     return;
```
Comment (lassoan) [inline Libs/MRML/Widgets/qMRMLDisplayNodeWidget.cxx]: indentation

**Comment by sjh26** [comment]
Indentation fixed!

**Comment by agirault** [comment]
Integrated in [r25731](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=25731). Thanks a lot Sam!

**Pull request closed without merging**